static/js/tools/map/chart_loader.tsx

```diff
@@ -151,133 +85,84 @@ export function ChartLoader(): JSX.Element {
 
+  // +++++++  BigQuery
+  // TODO: add webdriver test for BigQuery button to ensure query works
+  const getSqlQuery = useGetSqlQuery(chartStore);
+  const bqLink = useRef(setUpBqButton(getSqlQuery));
   useEffect(() => {
-    const placeSelected =
-      !_.isEmpty(placeInfo.value.enclosingPlace.dcid) &&
-      !_.isEmpty(placeInfo.value.enclosedPlaceType);
-    if (
-      placeSelected &&
-      !_.isEmpty(statVar.value.dcid) &&
-      !_.isNull(statVar.value.info)
-    ) {
-      fetchData(
-        placeInfo.value,
-        statVar.value,
-        isLoading,
-        setRawData,
-        display.value.showTimeSlider,
-        allSampleDates
-      );
-    } else {
-      setRawData(undefined);
-    }
-  }, [
-    placeInfo.value.enclosingPlace,
-    placeInfo.value.enclosedPlaceType,
-    statVar.value.dcid,
-    statVar.value.info,
-    statVar.value.denom,
-    statVar.value.mapPointSv,
-  ]);
-
+    const dom = bqLink.current;
+    dom.style.display = "none"; // Enable BQlink with "inline-block";
+    return () => {
+      dom.style.display = "none";
+    };
+  }, []);
+
+  // Set map type to leaflet if georaster data is available before data needed
+  // for d3 maps
   useEffect(() => {
-    // Set map type to leaflet if georaster data is available before data needed
-    // for d3 maps
     if (
-      (_.isEmpty(rawData) || _.isEmpty(chartStore.geoJson.data)) &&
+      (_.isEmpty(chartStore.mapValuesDates.data) ||
+        _.isEmpty(chartStore.geoJson.data)) &&
       !_.isEmpty(chartStore.geoRaster.data)
     ) {
       setMapType(MAP_TYPE.LEAFLET);
     }
-  }, [rawData, chartStore.geoJson.data, chartStore.geoRaster.data]);
-
-  useEffect(() => {
-    if (_.isEmpty(rawData) || _.isEmpty(chartStore.geoJson.data)) {
-      return;
-    }
-    const metahash = statVar.value.metahash || BEST_AVAILABLE_METAHASH;
-    const dateRawData =
-      statVar.value.date && sampleDatesChartData[metahash]
-        ? sampleDatesChartData[metahash][statVar.value.date]
-        : rawData;
-    loadChartData(
-      chartStore,
-      dateRawData,
-      placeInfo.value,
-      allSampleDates,
-      statVar.value,
-      setChartData,
-      dispatchSources,
-      dispatchMetadata,
-      display
-    );
-    // Should set legend bounds for per capita if there isn't a user specified
-    // domain.
-    if (
-      statVar.value.perCapita &&
-      statVar.value.denom &&
-      _.isEmpty(display.value.domain)
-    ) {
-      setLegendBoundsPerCapita(rawData, statVar, display);
-    }
   }, [
-    rawData,
+    chartStore.mapValuesDates.data,
     chartStore.geoJson.data,
     chartStore.geoRaster.data,
-    statVar.value.metahash,
-    statVar.value.perCapita,
   ]);
 
-  useEffect(() => {
-    if (onPlayCallback) {
-      onPlayCallback();
-    }
-  }, [sampleDatesChartData]);
+  // Rendering logic below ...
+  if (!statVar.value.info) {
+    return null;
+  }
+
+  if (!placeInfo.value.enclosingPlace.dcid || !placeInfo.value.enclosingPlace) {
```

Comment:
probably want to swap the order of the two checks?

Reply:
Moved all of these to a renderReady() checks now. In there these checks are not necessary, as it will check if computed values are ready.